File: seoyeon/week9/괄호 회전하기.py
def check(tmp) :
    stack = []
    for i in tmp :
        if len(stack) > 0 :
            if stack[-1] == '(' and i == ')' :
                stack.pop()
            elif stack[-1] == '{' and i == '}' :
                stack.pop()
            elif stack[-1] == '[' and i == ']' :
                stack.pop()
            else : stack.append(i)
        else : stack.append(i)
    if len(stack) == 0 :
        return 1
    else :
        return 0

def solution(s):
    answer = 0
    for i in range(len(s)) :
        if check(s):            # 체킹함수 실행 시 짝이 맞으면
            answer += 1         # answer 값 증가
        s = s[1:] + s[:1]       # 회전
    return answer

File: hyeyoung/week9/괄호회전하기.java
import java.util.*;


class Solution {
    public int solution(String s) {
        int answer = 0;
        String str=s;
        
        for(int i=0;i<s.length();i++){
            char temp = str.charAt(0);
            str = str.substring(1) + temp;
            if(checkString(str)){
                answer++;
            }
        }
        
        return answer;
    }
    
    public boolean checkString (String str){
        Stack<Character> s = new Stack<>();
        
        char[] arr = str.toCharArray();
        
        for(char data:arr){
            if(s.isEmpty()){
                s.push(data);
            }
            else if(s.peek()=='['&&data==']'){
                s.pop();
            }
            else if(s.peek()=='('&&data==')'){
                s.pop();
            }
            else if(s.peek()=='{'&&data=='}'){
                s.pop();
            }
            else{
                s.push(data);
            }
        }
        
        if(s.isEmpty()){
            return true;
        }
        
        return false;
    }
}
